src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,188 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MULTIPLIER_FOR_INCREASE_CAPACITY = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private int capacity;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        capacity = table.length;
+        threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        transferIfNeed();
+        int hashOfKey = calculateHash(key);
+        int bucketIndex = getBucketIndex(hashOfKey);
+        Node<K, V> existingNode = getNode(key);
+        if (existingNode != null) {
+            existingNode.value = value;
+        } else {
+            Node<K, V> newNode = new Node<>(hashOfKey, key, value, null);
+            insertNode(newNode, bucketIndex);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = getNode(key);
+        return node == null ? null : node.value;
+    }
+
+    private Node<K, V> getNode(K key) {
+        int hashOfKey = calculateHash(key);
+        int bucketIndex = getBucketIndex(hashOfKey);
+        if (table[bucketIndex] == null) {
+            return null;
+        }
+        Node<K, V> current = table[bucketIndex];
+        while (current != null) {
+            if (current.hash == hashOfKey
+                    && (current.key == key || Objects.equals(current.key, key))) {
+                return current;
+            }
+            current = current.next;
+        }
         return null;
     }
 
+    private void insertNode(Node<K, V> newNode, int bucketIndex) {
+        if (table[bucketIndex] == null) {
+            table[bucketIndex] = newNode;
+        } else {
+            getLastNode(bucketIndex).next = newNode;
```

Comment:
why do you need separate method for this if you are using it only in one please? 
methods should describe one particular finctionality. 
also if code duplicates we use separate method to avoid it